controllers/operator/configmap.go

```diff
@@ -265,6 +267,12 @@ func (r *AuthenticationReconciler) handleConfigMap(instance *operatorv1alpha1.Au
 					currentConfigMap.Data["PREFERRED_LOGIN"] = newConfigMap.Data["PREFERRED_LOGIN"]
 					cmUpdateRequired = true
 				}
+				if _, keyExists := currentConfigMap.Data["MASTER_HOST"]; !keyExists {
```

Comment:
Note that if you have the old ClusterCADomain value set from previous version, an upgrade will not replace this with the cluster_address given this check - the key exists in that case.

Reply:
```suggestion
				newConfigMap = functionList[index](instance, r.Scheme)
				if currentMasterHost, keyExists := currentConfigMap.Data["MASTER_HOST"]; !keyExists || currentMasterHost != newConfigMap.Data["MASTER_HOST"] {
```

Reply:
checking for keyExists